majilove/models.py

```diff
@@ -106,3 +108,402 @@ def display_blood_type(self):
+            'english': u'Dance',
+            'japanese_translation': u'DANCE'
+        }),
+        (2, {
+            'translation': _('Vocal'),
+            'english': u'Vocal',
+            'japanese_translation': u'VOCAL'
+        }),
+        (3, {
+            'translation': _('Charm'),
+            'english': u'Charm',
+            'japanese_translation': u'ACT'
+        })
+    ])
+
+    dance_min = models.PositiveIntegerField(string_concat(_('Dance'), ' (', _('Minimum'), ')'), default=0)
+    dance_single_copy_max = models.PositiveIntegerField(string_concat(_('Dance'), ' (', _('Single Copy Maximum'), ')'), default=0)
+    dance_max_copy_max = models.PositiveIntegerField(string_concat(_('Dance'), ' (', _('Maxed Copy Maximum'), ')'), default=0)
+
+    vocal_min = models.PositiveIntegerField(string_concat(_('Vocal'), ' (', _('Minimum'), ')'), default=0)
+    vocal_single_copy_max = models.PositiveIntegerField(string_concat(_('Vocal'), ' (', _('Single Copy Maximum'), ')'), default=0)
+    vocal_max_copy_max = models.PositiveIntegerField(string_concat(_('Vocal'), ' (', _('Maxed Copy Maximum'), ')'), default=0)
+
+    charm_min = models.PositiveIntegerField(string_concat(_('Charm'), ' (', _('Minimum'), ')'), default=0)
+    charm_single_copy_max = models.PositiveIntegerField(string_concat(_('Charm'), ' (', _('Single Copy Maximum'), ')'), default=0)
+    charm_max_copy_max = models.PositiveIntegerField(string_concat(_('Charm'), ' (', _('Maxed Copy Maximum'), ')'), default=0)
+
+    @property
+    def overall_min(self):
+        return self.dance_min + self.vocal_min + self.charm_min
+
+    @property
+    def overall_single_copy_max(self):
+        return self.dance_single_copy_max + self.vocal_single_copy_max + self.charm_single_copy_max
+
+    @property
+    def overall_max_copy_max(self):
+        return self.dance_max_copy_max + self.vocal_max_copy_max + self.charm_max_copy_max
+
+    # Leader Skill
+    LEADER_SKILL_INFO = {
+        'variables': ['color', 'stat', 'percentage'],
+        'template': _(u'{color} {stat} +{percentage}%'),
+        'japanese_template': u'{color} の{stat}パフォーマンス{percentage}%上昇'
+    }
+
+    #currently always the same as color, but this is safer
+    LEADER_SKILL_COLOR_CHOICES = COLOR_CHOICES
+    LEADER_SKILL_COLOR_WITHOUT_I_CHOICES = True
+    i_leader_skill_color = models.PositiveIntegerField('{color}', choices=LEADER_SKILL_COLOR_CHOICES, null=True)
+
+    LEADER_SKILL_STAT_CHOICES = [(_name, _info['translation']) for _name, _info in STATS.items()]
+    LEADER_SKILL_STAT_WITHOUT_I_CHOICES = True
+    i_leader_skill_stat = models.PositiveIntegerField('{stat}', choices=LEADER_SKILL_STAT_CHOICES, null=True)
+
+    leader_skill_percentage = models.PositiveIntegerField('{percentage}', null=True)
+
+    @property
+    def leader_skill_variables(self):
+        return {
+            key: getattr(self, u'leader_skill_{}'.format(key))
+            for key in self.LEADER_SKILL_INFO['variables']
+        }
```

Comment:
Can remove this function (see above + below for usage of new utility function`templateVariables`).